x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.ts

```diff
@@ -117,11 +121,18 @@ export const searchAfterAndBulkIndex = async (
 
   const totalHits =
     typeof someResult.hits.total === 'number' ? someResult.hits.total : someResult.hits.total.value;
-  let size = someResult.hits.hits.length - 1;
-  logger.debug(`first size: ${size}`);
+  // maxTotalHitsSize represents the total number of docs to
+  // query for. If maxSignals is present we will only query
+  // up to max signals - otherwise use the value
+  // from track_total_hits.
+  const maxTotalHitsSize = params.maxSignals ? params.maxSignals : totalHits;
+
+  // number of docs in the current search result
+  let hitsSize = someResult.hits.hits.length;
+  logger.debug(`first size: ${hitsSize}`);
   let sortIds = someResult.hits.hits[0].sort;
   if (sortIds == null && totalHits > 0) {
-    logger.error(`sortIds was empty on first search when encountering ${totalHits}`);
+    logger.warn('sortIds was empty on first search but expected more');
```

Comment:
Do you want this to be a `warn` even though below you are returning false to indicate an error?